### ISSUE TITLE
Add IrrlichtMt to minetest repo

### ISSUE THREAD

**Issue by NeroBurner** [minetest/minetest#11264]
## Problem

Starting with 5.5.0-dev minetest depends on a custom `IrrlichtMt` fork. So it is mandatory to provide those in some way, for example:

```
cmake -S . -B _build -DIRRLICHT_INCLUDE_DIR=/home/nero/repos/minetest/irrlicht/_install/include/irrlichtmt -DIRRLICHT_LIBRARY=/home/nero/repos/minetest/irrlicht/_install/lib/libIrrlichtMt.a -DRUN_IN_PLACE=ON
```

## Solutions

Either:
- copy the IrrlichtMt into a subdirectory to make it a bundled dependency like JsonCPP or LuaJIT
- add the IrrlichtMt repository as a git submodule

With both solutions one doesn't need to compile IrrlichtMt, install it to a random location, and then give that random location to the `minetest` with `IRRLICHT_INCLUDE_DIR` and `IRRLICHT_LIBRARY` cmake configure variables

**Comment by sfan5** [comment]
Both of your proposed solution have been discussed already and rejected.

Alternatively we should be looking at making it easy for users to build/use IrrlichtMt together with Minetest, some related discussion: https://irc.minetest.net/minetest-dev/2021-05-04#i_5817432

**Comment by NeroBurner** [comment]
so the solution is to use irrlicht from a subfolder at lib/irrlicht, if the user provides a copy of irrlichtmt in that directory?

**Comment by AntumDeluge** [comment]
I like the idea of being able to build irrlicht in parallel with Minetest. Configuring it as a dependency in `lib/irrlicht` sounds like a good idea, whether it be by Git module or manually downloading & extracting. As long as Minetest's CMake scripts can be configured to build it.

**Comment by ryvnf** [comment]
I consider this to be a duplicate of #11108 but I think that one can be closed instead since this one has more discussion.

I think allowing users to manually clone IrrlichtMt into `lib/irrlicht` and having the build script support that would be a good solution. I am also a bit curious to hear what the problems with submodules are. I read [the IRC discussion](https://irc.minetest.net/minetest/2021-03-31#i_5806055) and many people seemed to agree they suck but I did not find reasons why.
